ZenPacks/zenoss/Import4/rrd2tsdb.py

```diff
@@ -91,25 +103,26 @@ def _process_v(self, tag_path, content):
         if tag_path != '/rrd/rra/database/row/v':
             return
 
-        # TBD need to do different calc for different type
-        if self.type == 'GAUGE':
-            if content.strip() != 'NaN':
-                print '%s %s %s device=%s key=%s zenoss_tenant_id=%s' % (
-                    self.metric, self.last_timestamp, content.strip(),
-                    self.device, self.key, self.dmd_uuid)
-        elif self.type == 'COUNTER':
-            log.warning("COUNTER NOT HANDLED YET")
-            return
-        elif self.type == 'DERIVE':
-            log.warning("DERIVE NOT HANDLED YET")
-            return
+        # don't output NaN values
+        if content.strip() == 'NaN':
             return
+
+        # compute the tsdb data value per type
+        if self.type == 'GAUGE':
+            _value = float(content.strip())
+        elif self.type == 'DERIVE' or self.type == 'COUNTER':
+            self.derived_value += float(content.strip()) * self.step
+            _value = self.derived_value
```

Comment:
Does self.step ever change? If not, we could accumulate less rounding error by doing this instead:

            self.derived_value += float(content.strip())
            _value = self.derived_value * self.step